### PR TITLE
Fix elementwise arithmetic with zero sized dimensions

### LINKED ISSUE #893: torch.trapz with empty dim
```
import torch
import torch_xla

device = 'xla:0'
# device='cpu'

sizes = (0, 2)
dim=1
x=[1.0, 2.0]

t = torch.randn(sizes, device=device)

a = torch.trapz(t, x=torch.tensor(x, device=device), dim=dim)
print(a)
```
Expected: should match cpu result.

### PR DESCRIPTION
**Pull request by asuhan**
Fixes #893.

### CODE REVIEW

File: torch_xla/csrc/helpers.cpp [hunk 1]
```diff
@@ -307,7 +307,11 @@ xla::Shape XlaHelpers::GetPromotedShape(const xla::Shape& shape1,
     xla::int64 dim2 = shape2_dims[shape2_dims.size() - min_size + i];
     XLA_CHECK(dim1 == dim2 || dim1 == 1 || dim2 == 1)
         << shape1 << " and " << shape2;
-    dimensions.push_back(std::max<xla::int64>(dim1, dim2));
+    if (dim1 == 0 || dim2 == 0) {
```
Comment (dlibenzi) [inline torch_xla/csrc/helpers.cpp]: I fear with all these letting through zero-sized dimensions, to be ending up chasing a lot of places to prevent them to hit XLA.
I hope I am wrong.

Comment (asuhan) [reply torch_xla/csrc/helpers.cpp]: I partially share your concern. In principle, though, XLA supports zero-sized dimensions so we should be fine.
